python/ray/ml/config.py

```diff
@@ -35,29 +34,71 @@ class ScalingConfigDataClass:
         Strategies <pgroup-strategy>` for the possible options.
     """
 
-    num_workers: int
+    num_workers: int = 1
     use_gpu: bool = False
     resources_per_worker: Optional[Dict] = None
     placement_strategy: str = "PACK"
 
+    def __post_init__(self):
+        self.resources_per_worker = (
+            self.resources_per_worker if self.resources_per_worker else {}
+        )
+
+        if self.num_workers < 0:
```

Comment:
What about single-threaded trainers, isn't that like having 0 workers?

Reply:
Got it ok- I'll remove this check to support those use cases.

Though none of the Trainers that we currently have support a local mode.

Reply:
Removed!